src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -322,6 +322,19 @@ internal async Task<ImmutableArray<ActiveStatementExceptionRegions>> GetBaseActi
                     continue;
                 }
 
+                // Check if the currently observed document content has changed compared to the base document content.
+                // This is an important optimization that aims to avoid IO while stepping in sources that have not changed.
+                // We may be comparing out-of-date committed document content but we only make a decision based on that content
+                // if it matches the current content. That means the change on disk has not been observed in the workspace yet
+                // and the buffer is showing the out-of-date content, if the document is opened. The unobserved change is not
+                // gonna be applied at this moment. It might be applied later after it's observed.
+                var baseSource = await baseProject.GetDocument(documentId)!.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var source = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                if (baseSource.ContentEquals(source))
```

Comment:
On line 317 you can also pass "onlyGetDocumentsWithTextChanges: true" to GetChangedDocuments to filter out documents that absolutely haven't changed the text of. I'll warn that it only filters out things where the underlying "text" object is reference equal, so you may still want this code as well, but that'll avoid you even having to fetch text and call ContentEquals if not necessary.

Reply:
Oh, I didn't know that option exists. I'll use that. Thanks